File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/mergetwo2darraysbysummingvalues/Solution.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        ret = []
        i, j = 0, 0
        len1, len2 = len(nums1), len(nums2)

        while i < len1 or j < len2:
            if i < len1 and j < len2:
                if nums1[i][0] == nums2[j][0]:
                    ret.append([nums1[i][0], nums1[i][1] + nums2[j][1]])
                    i += 1
                    j += 1
                elif nums1[i][0] < nums2[j][0]:
                    ret.append([nums1[i][0], nums1[i][1]])
                    i += 1
                else:
                    ret.append([nums2[j][0], nums2[j][1]])
                    j += 1
            elif i < len1:
                ret.append([nums1[i][0], nums1[i][1]])
                i += 1
            else:
                ret.append([nums2[j][0], nums2[j][1]])
                j += 1

        return ret

File: algorithm-problems/src/main/java/us/supercheng/algorithm/problems/leetcode/mergetwo2darraysbysummingvalues/Solution.java
package us.supercheng.algorithm.problems.leetcode.mergetwo2darraysbysummingvalues;

import java.util.ArrayList;
import java.util.List;

class Solution {

    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        List<int[]> list = new ArrayList<>();

        for (int i = 0, j = 0, len1 = nums1.length, len2 = nums2.length; i < len1 || j < len2; ) {
            if (i < len1 && j < len2) {
                if (nums1[i][0] == nums2[j][0])
                    list.add(new int[]{nums1[i][0], nums1[i++][1] + nums2[j++][1]});
                else if (nums1[i][0] < nums2[j][0])
                    list.add(new int[]{nums1[i][0], nums1[i++][1]});
                else
                    list.add(new int[]{nums2[j][0], nums2[j++][1]});
            } else if (i < len1)
                list.add(new int[]{nums1[i][0], nums1[i++][1]});
            else
                list.add(new int[]{nums2[j][0], nums2[j++][1]});
        }

        int[][] ret = new int[list.size()][2];
        for (int i = 0, len = list.size(); i < len; i++)
            ret[i] = list.get(i);

        return ret;
    }

}
